Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -237,7 +237,7 @@ private async void OnPageSizeChanged( int pageSize )
 
             if ( ManualReadMode )
             {
-                await InvokeAsync( () => HandleReadData( paginationContext.CancellationTokenSource.Token ) );
+                await InvokeAsync( () => HandleReadData( -1, paginationContext.CancellationTokenSource.Token ) );

```

Comment:
Do we need to introduce -1?

Reply:
No we don't... I kinda just did random test code on that commit 😆
However it might be have to be the default as start idx 0 should be valid.